pkg/api/validation/validation.go

```diff
@@ -373,9 +398,11 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.Namespace, oldMeta.Namespace, fldPath.Child("namespace"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.UID, oldMeta.UID, fldPath.Child("uid"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newMeta.CreationTimestamp, oldMeta.CreationTimestamp, fldPath.Child("creationTimestamp"))...)
+	allErrs = append(allErrs, ValidateImmutableField(newMeta.Finalizers, oldMeta.Finalizers, fldPath.Child("finalizers"))...)
```

Comment:
The cascading escalation surface area is significant.  The power to update foo gives me the power to delete not only foo, but I'm escalated to delete everything depending on foo.

I don't think most people expect that as a side effect of update powers.

Also, the fact that the API behaves in a way that defies @liggitt's expectation may be an indication that it isn't obeying the principle of least surprise.